src/Classes/ProfilePage.php

```diff
@@ -0,0 +1,1020 @@
+			}
+			return;
+		}
+
+		$profileTitle = $this->isProfilePage( $title ) ? $title : Title::makeTitle( NS_USER_PROFILE, $userName );
+		$userPageTitle = $this->isUserPage( $title ) ? $title : Title::makeTitle( NS_USER, $title->getDBKey() );
+		$userTalkPageTitle = $this->isUserTalkPage( $title ) ? $title :
+			Title::makeTitle( NS_USER_TALK, $title->getDBKey() );
+
+		$links['namespaces'] = [];
+
+		if ( $this->isProfilePage( $title ) ) {
+			// Reset $links to prevent hidden article.
+			$links['namespaces'] = [];
+			$links['views'] = [];
+			$links['actions'] = [];
+			$links['variants'] = [];
+		}
+
+		// Build Link for Profile
+		$links['namespaces']['userprofile'] = [
+			'class' => $this->isProfilePage( $title ) ? 'selected' : '',
+			'href' => $profileTitle->getFullURL(),
+			'text' => wfMessage( 'userprofiletab' )->text(),
+			'primary' => true
+		];
+
+		// Build Link for User Page
+		$class = [];
+		if ( $this->isUserPage( $title ) ) {
+			$class[] = 'selected';
+		}
+		if ( !$userPageTitle->isKnown() ) {
+			$class[] = 'new';
+		}
+		$links['namespaces']['user'] = [
+			'class' => implode( ' ', $class ),
+			'text' => wfMessage( 'nstab-' . $userPageTitle->getNamespaceKey( '' ) )->text(),
+			'href' => $this->profile->getUserPageUrl( $userPageTitle ),
+			'primary' => true
+		];
+
+		// Build Link for User Talk Page
+		$class = [];
+		if ( $this->isUserTalkPage( $title ) ) {
+			$class[] = 'selected';
+		}
+		if ( !$userTalkPageTitle->isKnown() ) {
+			$class[] = 'new';
+		}
+		$links['namespaces']['user_talk'] = [
+			'class' => implode( ' ', $class ),
+			'text' => wfMessage( 'talk' )->text(),
+			'href' => $this->profile->getTalkPageUrl( $userTalkPageTitle ),
+			'primary' => true
+		];
+
+		$links['views']['contribs'] = [
+			'class' => false,
+			'text' => wfMessage( 'contributions' )->text(),
+			'href' => SpecialPage::getTitleFor( 'Contributions', $userName )->getFullURL(),
+		];
+	}
+
+	/**
+	 * Gets an md5 hash for gravatar URLs
+	 *
+	 * @param string $email User's email address
+	 * @return string md5 hash of the email address
+	 */
+	private static function emailToMD5Hash( $email ) {
+		return md5( strtolower( trim( $email ) ) );
+	}
+
+	/**
+	 * Prints a gravatar image tag for a user
+	 *
+	 * @param null $parser - Not Used but passed by MW
+	 * @param int $size the square size of the avatar to display
+	 * @param string $email email Address OR md5 Hash of user's email address
+	 * @param string $userName the user's username
+	 * @param string $attributeString additional html attributes to include in the IMG tag
+	 * @return array the HTML fragment containing a IMG tag
+	 */
+	public static function userAvatar( $parser, $size = 32, $email = '', $userName = '', $attributeString = '' ) {
+		if ( empty( $email ) ) {
+			return [ '', 'isHTML' => true ];
+		}
+
+		$size = (int)$size;
+		$userName = htmlspecialchars( $userName, ENT_QUOTES );
+		$attributeString = htmlspecialchars( $attributeString, ENT_QUOTES );
```

Comment:
Fix backported in https://github.com/Wikia/mediawiki-extensions-CurseProfile/pull/30/files